src/main/kotlin/org/rust/ide/inspections/RsAssertEqualInspection.kt

```diff
@@ -0,0 +1,86 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections
+
+import com.intellij.codeInspection.LocalQuickFix
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.ComparisonOp
+import org.rust.lang.core.psi.ext.macroName
+import org.rust.lang.core.psi.ext.operatorType
+
+class RsAssertEqualInspection : RsLocalInspectionTool() {
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean) = object : RsVisitor() {
+
+        override fun visitMacroCall(o: RsMacroCall) {
+            val macroName = o.macroName?.text ?: return
+            val assertMacroArg = o.assertMacroArgument ?: return
+            if (macroName != "assert") return
+
+            val expr = assertMacroArg.expr as? RsBinaryExpr ?: return
+            when (expr.operatorType) {
+                ComparisonOp.EQ -> {
+                    holder.registerProblem(
+                        o,
+                        "assert!(a == b) can be assert_eq!(a, b)",
+                        SpecializedAssertQuickFix("assert_eq")
+                    )
+                }
+                ComparisonOp.EXCLEQ -> {
+                    holder.registerProblem(
+                        o,
+                        "assert!(a != b) can be assert_ne!(a, b)",
+                        SpecializedAssertQuickFix("assert_ne")
+                    )
+                }
+                else -> return
+            }
+        }
+    }
+
+
+    private class SpecializedAssertQuickFix(private val assert_name: String) : LocalQuickFix {
+        override fun getName() = "Convert to %s!".format(assert_name)
```

Comment:
You can use string interpolation instead of `format` method: `"Convert to $assert_name!"`